coder_sniffer/Drupal/Sniffs/Classes/ClassFileNameSniff.php

```diff
@@ -68,6 +68,17 @@ public function process(File $phpcsFile, $stackPtr)
             $fileName = str_replace('_', '', ucwords($fileName, '_')).'SubContext';
         }
 
+        // If the file is either txt or md, it probably should not be named
+        // after the class we are searching for.
+        $extension         = pathinfo($fullPath, PATHINFO_EXTENSION);
+        $ignoredExtensions = [
```

Comment:
Now that I think about it - I think we should not have a list of extensions to ignore, we should reverse this and only accepts *.php files.

Can you change that and move this check further up? Right after where we check for empty/no file name.

Reply:
I was thinking we could also include other known php files (like .module, .inc and .theme) since the fact that you are declaring a class in there might be a code smell. I updated the code with that, what do you think?

Reply:
...and also the fact that we have code specifically handling .behat.inc files in both text fixtures and in the sniff class :)